packages/parser/src/language/sankey/sankeyMatcher.ts

```diff
@@ -0,0 +1,91 @@
+import { accessibilityDescrRegex, accessibilityTitleRegex, titleRegex } from '../common/matcher.js';
+
+import type {
+  CustomPatternMatcherFunc,
+  CustomPatternMatcherReturn,
+  IToken,
+} from '../chevrotainWrapper.js';
+
+/**
+ * Matches sankey link source and target node
+ */
+export const sankeyLinkNodeRegex = /(?:"((?:""|[^"])+)")|([^\n\r,]+(?=%%)|[^\n\r,]+)/;
```

Comment:
I know this line originates from #4799, nevertheless I think we have to pre-process such things as `%%` so as not to include them in grammar / lexers at all if possible. Removing these lines beforehand will reduce overall complexity

Reply:
totally agree.

Reply:
%% lines are already removed by mermaid before parsing.

Reply:
The parser should handle comments in general, it should be a standalone package.
I'm not sure if we should have that pre-process, but if we do, then it should be in the parser.

But how accurate it is? Some places it's possible to have `%%` in general. For example:

```
pie
"section
%% not comment
name": 1.0
```

Reply:
I agree, if the new parser can handle the comments in a common manner, it should do it. 
We're already facing issues with incorrect line numbers in errors and stuff.